instrumentation/events/lttng-module/net.h

```diff
@@ -258,6 +412,14 @@ LTTNG_TRACEPOINT_EVENT_CLASS(net_dev_template,
 					/* For any other header type, there is nothing to do. */
 					break;
 				}
+
+				/* Copy the transport header */
+				if (__get_transport_header_type(skb) == TH_TCP) {
+					ctf_array_type(uint8_t, tcp_hdr(skb),
+							sizeof(struct tcphdr))
+					ctf_align(uint8_t)
```

Comment:
ref http://diamon.org/ctf/

4.2.2 Variants (discriminated/tagged unions)

"The alignment of the type containing the variant is independent of the variant alignment. For instance, if a structure contains two fields, a 32-bit integer, aligned on 32 bits, and a variant, which contains two choices: either a 32-bit field, aligned on 32 bits, or a 64-bit field, aligned on 64 bits, the alignment of the outmost structure will be 32-bit (the alignment of its largest field, disregarding the alignment of the variant). The alignment of the variant will depend on the selector: if the variant’s 32-bit field is selected, its alignment will be 32-bit, or 64-bit otherwise. It is important to note that variants are specifically tailored for compactness in a stream. Therefore, the relative offsets of compound type fields can vary depending on the offset at which the compound type starts if it contains a variant that itself contains a type with alignment larger than the largest field contained within the compound type. This is caused by the fact that the compound type may contain the enumeration that select the variant’s choice, and therefore the alignment to be applied to the compound type cannot be determined before encountering the enumeration."

So the alignment of the network header is independent of the TCP header alignment, because the tcp header will be in a variant.

Since all fields are naturally aligned, it should not require that we "pack" them (8-bit alignment), so I don't see why we should enforce that.